src/main/java/edu/harvard/iq/dataverse/ThumbnailServiceWrapper.java

```diff
@@ -49,15 +49,14 @@ public class ThumbnailServiceWrapper implements java.io.Serializable  {
     private Map<Long, Boolean> hasThumbMap = new HashMap<>();
 
     public String getFileCardImageAsUrl(SolrSearchResult result) {
-        if (result.isHarvested()) {
-            return null;
-        }
 
-        if (result.getEntity() == null) {
-            return null;
+        if (!result.isHarvested() && result.getEntity() != null && (!((DataFile)result.getEntity()).isRestricted()
```

Comment:
This should probably use the logic in https://github.com/IQSS/dataverse/blob/9a1e49453731ffc57f4dc13fdea88e6728957ff2/src/main/java/edu/harvard/iq/dataverse/api/Access.java#L1759 or https://github.com/IQSS/dataverse/blob/9a1e49453731ffc57f4dc13fdea88e6728957ff2/src/main/java/edu/harvard/iq/dataverse/DatasetPage.java#L5831-L5836 to address embargo and retention

Reply:
@qqmyers Wouldn't the Access call using  the image_url go through the isAccessAuthorized() method anyways? So if we supply the url the user then needs to make the call to get the image which would fail is they don't have access. I don't want to duplicate the code so I would need to make it public and be able to call it from the wrapper to prevent sending the url.

Reply:
Yes, and it would make sense to change the code to public or move the relevant parts to a util class to avoid duplicating code.

Reply:
@qqmyers I went with the second option of copying the code from DatasetPage